File: menuRestApi/src/main/java/com/kh/menu/model/vo/Menu.java
package com.kh.menu.model.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Menu {
	private int id;
	private String restaurant;
	private String name;
	private int price;
	private MenuType type;
	private MenuTaste taste;
}

File: menuapp/src/type/menu.ts
export type MenuType = 'all'|'kr'|'jp'|'ch';
export type MenuTaste = 'all'|'mild'|'hot';

export interface Menu{
    id : number,
    restaurant : string,
    name : string,
    price : number,
    type : MenuType,
    taste : MenuTaste
}

export const initialMenu:Menu = {
    id : 0,
    restaurant : '',
    name : '',
    price : 0,
    type : 'kr',
    taste : 'mild'
} as const; 
// 값이 바뀔수 있으므로 (공통으로 사용?) const 지정
// (리터럴 + readonly)

export const initialMenuList:Menu[] = [];

export type SearchKeyword = {
    type:MenuType;
    taste: MenuTaste;
}